src/main/scala/PropertyChecker.scala

```diff
@@ -630,4 +664,285 @@ object PropertyChecker {
+    }
+  }
+
+  def naturalLanguageProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      var warnings = Array[String]()
+      value match {
+        case s:TextNode => {
+          val returnObject = JsonNodeFactory.instance.objectNode()
+          val arrayNode = JsonNodeFactory.instance.arrayNode()
+          arrayNode.add(s.asText())
+          returnObject.set("lang", arrayNode)
+          (returnObject, Array[String](), csvwPropertyType)
+        }
+        case a:ArrayNode => {
+          var validTitles = Array[String]()
+          val arrayNodeIterator = a.elements()
+          while(arrayNodeIterator.hasNext()) {
+            val element = arrayNodeIterator.next()
+            element match {
+              case s:TextNode => validTitles = validTitles :+ s.asText()
+              case _ => warnings = warnings :+ PropertyChecker.invalidValueWarning
+            }
+          }
+          val returnObject = JsonNodeFactory.instance.objectNode()
+          val arrayNode:ArrayNode = PropertyChecker.mapper.valueToTree(validTitles)
+          returnObject.set("lang", arrayNode)
+          (returnObject, warnings, csvwPropertyType)
+        }
+        case o:ObjectNode => {
+          val valueCopy = o.deepCopy()
+          val fieldsAndValues = Array.from(valueCopy.fields.asScala)
+          for(fieldsAndValue <- fieldsAndValues) {
+            val elementKey = fieldsAndValue.getKey
+            val matcher = PropertyChecker.Bcp47LanguagetagRegExp.pattern.matcher(elementKey)
+            if(matcher.matches()) {
+              var validTitles = Array[String]()
+              fieldsAndValue.getValue match {
+                case s:TextNode => validTitles = validTitles :+ s.asText()
+                case a:ArrayNode => {
+                  val titles = Array.from(a.elements().asScala)
+                  for(title <- titles) {
+                    title match {
+                      case s:TextNode => validTitles = validTitles :+ s.asText()
+                      case _ => warnings = warnings :+ PropertyChecker.invalidValueWarning
+                    }
+                  }
+                }
+                case _ => warnings = warnings :+ PropertyChecker.invalidValueWarning
+              }
+              val validTitlesArrayNode:ArrayNode = PropertyChecker.mapper.valueToTree(validTitles)
+              valueCopy.set(elementKey, validTitlesArrayNode)
+            } else {
+              valueCopy.remove(elementKey)
+              warnings = warnings :+ "invalid_language"
+            }
+          }
+          (valueCopy, warnings, csvwPropertyType)
+        }
+        case _ => (NullNode.getInstance(), Array[String](PropertyChecker.invalidValueWarning), csvwPropertyType)
+      }
+    }
+  }
+
+  def nameProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
```

Comment:
Is the `csvwPropertyType` param used here?

Reply:
Directly returned the type in some methods. Now I have changed them and I am using the passed in type (both are same, but now it uses the declared csvwPropertyType variable)